### ISSUE TITLE
Support for dynamic/hard-coded keys in --unused check

### ISSUE THREAD

**Issue by ceolinwill** [lingualdev/i18n-check#89]
Right now, the `--unused` check only works when keys are used directly in `t("key")`. If keys are passed dynamically, they are incorrectly reported as unused.

For example:

```ts
const menuItems = [
  { i18nKey: "home" }
];

menuItems.map((item) => t(item.i18nKey));
```

In this case, the `home` key is marked as unused, even though it is clearly used in the app.

**Proposal:**

It would be great if `i18n-check` could also detect dynamic key usage. I understand full static analysis may be tricky, but perhaps a simpler workaround could be added.

For example, we could introduce a flag to tell the tool which object property to check when resolving dynamic keys, something like:

```
--unused-key i18nKey
```

This would make the `unused` check also look for values of `i18nKey` properties across the codebase.

**Comment by busypeoples** [comment]
One issue, that I can see with this approach is that sometimes these values could be loaded via an API or from a JSON file at runtime. What do you think about defining these dynamic keys as comments like in the `18next` [example](https://github.com/lingualdev/i18n-check/blob/main/translations/codeExamples/reacti18next/src/App.tsx#L21)?
The `next-intl` parser should be able to detect comments already or can be extended to do that.

**Comment by ceolinwill** [comment]
Oh, yeah. That works, thanks for the tip!
